pkg/util/limit/limiter.go

```diff
@@ -52,6 +54,7 @@ func (l *ConcurrentRequestLimiter) Begin(ctx context.Context) (Reservation, erro
 		var span *tracing.Span
 		ctx, span = tracing.ChildSpan(ctx, l.spanName)
 		defer span.Finish()
+		span.RecordStructured(&types.StringValue{Value: fmt.Sprintf("%d requests are waiting in the pool", l.sem.Len())})
```

Comment:
nit: s/in the pool// (also, I could see this file being separate commit)

Reply:
done.